api/app/events/models.py

```diff
@@ -12,14 +12,50 @@ class Event(db.Model):
     start_date = db.Column(db.DateTime(), nullable=False)
     end_date = db.Column(db.DateTime(), nullable=False)
 
+    application_open = db.Column(db.DateTime(), nullable=False)
+    appcliation_close = db.Column(db.DateTime(), nullable=False)
+    review_open = db.Column(db.DateTime(), nullable=False)
+    review_close = db.Column(db.DateTime(), nullable=False)
+    selection_open = db.Column(db.DateTime(), nullable=False)
+    selection_close = db.Column(db.DateTime(), nullable=False)
+    offer_open = db.Column(db.DateTime(), nullable=False)
+    offer_close = db.Column(db.DateTime(), nullable=False)
+    registration_open = db.Column(db.DateTime(), nullable=False)
+    registration_close = db.Column(db.DateTime(), nullable=False)
+
```

Comment:
These fields all look good however we're still missing the migration that will generate the script that will add these columns to the database when it's time for us to deploy. So once you fix the typos, run the following command

`docker-compose run web python ./api/run.py db migrate --directory api/migrations`

Assuming no errors occurred (generally this will happen if there's a syntax error in your code), this should generate the script. If you look in your project directory, you will see a new file has been added to `./app/migrations/versions`. You'll also see all the other files that have been generated previously.

Open the file and verify that in the `upgrade()` method the columns have been added and in the `downgrade()` method, the columns have been removed. The` upgrade()` is for when we deploy and the `downgrade()` is for if we decide to rollback a change on prod.

Once you're happy with the script, run the following command,

`docker-compose run web python ./api/run.py db upgrade --directory api/migrations`

This should now actually run the script that was generated and apply the changes to your local instance of your database. Pesonally I use [pgAdmin ](https://www.pgadmin.org/download/) as a sanity check to see in the database itself in a nice GUI if the changes I expected are actually there. 

Also in case you need to reference these commands again in the future, have a look at the `./api/README.md`. I know I still look at it all the time :)

Reply:
Thank you very much for this detailed review. Appreciate it. I have made the changes. :)

Reply:
@amritpurshotam  I believe a lot of your above replies can be incorporated in the development docs for newcomers and everyone else. @avishkar58